src/DevTools/DevTools.php

```diff
@@ -55,6 +55,25 @@ public function onCommand(CommandSender $sender, Command $command, string $label
 			case "makeplugin":
 				if(isset($args[0]) and $args[0] === "FolderPluginLoader"){
 					return $this->makePluginLoader($sender, $command, $label, $args);
+				}elseif(isset($args[0]) and $args[0] === "*"){
+					$plugins = $this->getServer()->getPluginManager()->getPlugins();
+					$succeeded = $failed = [];
+					foreach($plugins as $plugin){
```

Comment:
this should check if the loader is a `FolderPluginLoader` instance before trying to build it.
```
makeplugin *
[10:09:58] [Server thread/INFO]: Plugin DevTools is not in folder structure.
[10:09:58] [Server thread/INFO]: [DevTools] Adding files...
[10:09:58] [Server thread/INFO]: [DevTools] Added 4 files
[10:09:58] [Server thread/INFO]: [DevTools] Checking for compressible files...
[10:09:58] [Server thread/INFO]: [DevTools] Done in 0.427s
[10:09:58] [Server thread/INFO]: Phar plugin DumpBadLogins v0.0.1 has been created on plugins\DevTools/\DumpBadLogins_v0.0.1.phar
[10:09:58] [Server thread/INFO]: [DevTools] Adding files...
[10:09:58] [Server thread/INFO]: [DevTools] Added 2 files
[10:09:58] [Server thread/INFO]: [DevTools] Checking for compressible files...
[10:09:58] [Server thread/INFO]: [DevTools] Done in 0.041s
[10:09:58] [Server thread/INFO]: Phar plugin FloatingText v0.0.1 has been created on plugins\DevTools/\FloatingText_v0.0.1.phar
[10:09:58] [Server thread/INFO]: [DevTools] Adding files...
[10:09:58] [Server thread/INFO]: [DevTools] Added 2 files
[10:09:58] [Server thread/INFO]: [DevTools] Checking for compressible files...
[10:09:58] [Server thread/INFO]: [DevTools] Done in 0.054s
[10:09:58] [Server thread/INFO]: Phar plugin test v0.0.1 has been created on plugins\DevTools/\test_v0.0.1.phar
[10:09:58] [Server thread/INFO]: 1 plugin failed to build: DevTools
[10:09:58] [Server thread/INFO]: 3/4 plugins successfully built: DumpBadLogins, FloatingText, test
```

Reply:
since makeplugin doesn't check either I left it as it was - I was more concerned about whether to exclude DevTools itself, or leave it for informational purposes.

Edit - I mean that this is checked within makePluginCommand(), so I didn't also add a check for within onCommand(). I've added the check in the latest commit, but this simply silences the error "Plugin DevTools is not in folder structure."